pulser/waveforms.py

```diff
@@ -622,8 +642,10 @@ def __getitem__(self, index_or_slice:
             self._check_slice(index_or_slice)
             return cast(np.ndarray, self.samples[index_or_slice])
         else:
-            self._check_index(index_or_slice)
-            return cast(float, self.samples[index_or_slice])
+            index: int = self._check_index(index_or_slice)
+            if index == 0 or index == self._duration - 1:
+                return 0.
+            return cast(float, self.samples[index])
```

Comment:
You can do something similar on `RampWaveform` to return `start` and `stop`.

Reply:
:+1: